### ISSUE TITLE
The /freecar command should attempt to place a car in vacant spaces, not haphazardly

### ISSUE THREAD

**Issue by ThePiachu** [StrangeLoopGames/EcoSuggestions#1105]
While the /freecar command is pretty useful, using it on narrow bridges or in tunnels has about 50+% chance of making your problem even worse:

![image](https://user-images.githubusercontent.com/964559/80913622-d024b280-8cfa-11ea-8f39-569bac4d0687.png)

It would be nice if the command at least attempted to put the car in an open spot. Say it would check the surrounding area for a 2x3x4 spot of empty air and place the car in that slot, orientated correctly. This would make the command a lot more useful in situations other than open fields.

**Comment by trickenso** [comment]
I don't believe the function is haphazard - it appears to place the vehicle at a right angle relative to the player, which allows you to solve most of these placement problems by moving yourself around.

Agreed that it might be more helpful to detect a suitable flat area (or at least one unoccupied by blocks) and place the vehicle there. Especially useful in narrow tunnels where you can't easily reach a position perpendicular to the ideal position of the vehicle.

Some additional explanation of exactly how this command works might also help manage expectations.
